app/src/main/java/com/github/fribourgsdp/radio/JoinWithQRCodeFragment.kt

```diff
@@ -0,0 +1,165 @@
+
+            override fun onScannerError(err: Throwable) {
+                // method is called when scanner gives some error.
+                Toast.makeText(ctx,"Scanner Error: " + err.message,Toast.LENGTH_SHORT).show()
+            }
+
+            override fun onCodeScanned(data: String) {
+                // method is called when camera scans the
+                // qr code and the data from qr code is
+                // stored in data in string format.
+                actOnScannedData(data)
+
+            }
+        }
+    }
+
+    protected fun actOnScannedData(data :String){
+        scannedTV.setText(data)
+        if(checkScanFormat(data)){
+            setFragmentResult("idRequest", bundleOf("id" to data.toLong()))
+            dismiss()
+        }
+    }
+
+    private fun checkScanFormat(data: String) : Boolean{
+        return try{
+            data.toLong()
+            true
+        }catch(e : NumberFormatException){
+            Toast.makeText(ctx,"False format",Toast.LENGTH_SHORT).show()
+            false
+        }
+
+
+    }
+
+    override fun onResume() {
+        super.onResume()
+        val decoder = ZXDecoder()
+        // 0.5 is the area where we have
+        // to place red marker for scanning.
+        decoder.scanAreaPercent = 0.8
+        // below method will set secoder to camera.
+        camera.decoder = decoder
+        camera.startScanner()
+    }
+
+    override fun onPause() {
+        // on app pause the
+        // camera will stop scanning.
+        camera.stopScanner()
+        super.onPause()
+    }
+
+    private fun checkPermission(): Boolean {
+        // here we are checking two permission that is vibrate
+        // and camera which is granted by user and not.
+        // if permission is granted then we are returning
+        // true otherwise false.
+        val camera_permission = ContextCompat.checkSelfPermission(ctx,
+            Manifest.permission.CAMERA
+        )
+        val vibrate_permission = ContextCompat.checkSelfPermission(ctx,
+            Manifest.permission.VIBRATE
+        )
+        return camera_permission == PackageManager.PERMISSION_GRANTED && vibrate_permission == PackageManager.PERMISSION_GRANTED
+    }
+
+
+    protected open fun requestPermission() {
+        // this method is to request
+        // the runtime permission.
+        val PERMISSION_REQUEST_CODE = 200
+        ActivityCompat.requestPermissions(activity, arrayOf(
+            Manifest.permission.CAMERA,
+            Manifest.permission.VIBRATE
+        ), PERMISSION_REQUEST_CODE)
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        // this method is called when user
+        // allows the permission to use camera.
+        if (grantResults.size > 0) {
+            val cameraaccepted = grantResults[0] == PackageManager.PERMISSION_GRANTED
```

Comment:
I think you should rename the variable name with as cameraAccepted. Same for below variables

Reply:
yes indeed